client/src/components/Join.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import io from 'socket.io-client';
+
+const url = 'http://localhost:8080';
+const socket = io(url);
+
+const Join = ({ setTopLevelState }) => {
+  const [name, setName] = useState('');
+
+  const onChange = (event) => {
+    setName(event.target.value);
+  };
+
+  // Event handlers would need to ..
```

Comment:
These comments seem useful to guide future work when connecting with DB so these look good to stay